src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/TestRouter.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2022 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.feedzai.commons.sql.abstraction.engine.impl.abs;
+
+import com.feedzai.commons.sql.abstraction.engine.DatabaseEngine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.JDBC;
+
+/**
+ * A class that acts as a router, to forward data sent between a {@link DatabaseEngine} and the DB.
+ * <p>
+ * When running tests, this offers the possibility of interrupting connections without closing them.
+ *
+ * @author José Fidalgo (jose.fidalgo@feedzai.com)
+ */
+public class TestRouter implements Closeable {
```

Comment:
Why not make this `AutoCloseable`?

Reply:
Closeable extends AutoCloseable, it is also auto-closed when used in try-with-resources. 
Usually I refer to the documentation, where it says for AutoCloseable:

> Note that unlike the {@link java.io.Closeable#close close} method of {@link java.io.Closeable}, this {@code close} method is <em>not</em> required to be idempotent.

And for Closeable:
> If the stream is already closed then invoking this method has no effect.

I want to explicitly state that the TestRouter#close behaves like the one in Closeable.